dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -71,15 +70,22 @@ object Datadog {
             )
 
             if (sdkInstanceId == null) {
-                internalLogger.log(
+                unboundInternalLogger.log(
                     InternalLogger.Level.ERROR,
                     InternalLogger.Target.USER,
                     CANNOT_CREATE_SDK_INSTANCE_ID_ERROR
                 )
                 return null
             }
 
-            val sdkCore = DatadogCore(context, credentials, configuration, sdkInstanceId)
+            val sdkCore = DatadogCore(
+                context,
+                credentials,
+                sdkInstanceId,
+                instanceName ?: SdkCoreRegistry.DEFAULT_INSTANCE_NAME
```

Comment:
Maybe we could extract this in a local var `sdkInstanceName` and reuse it in the `registry.register(instanceName, sdkCore)` below. 
Right now the implementation ensures that the `instanceName`, if null, will be converted to the same `DEFAULT_INSTANCE_NAME`, but it might cause a bug in the future if we change the registry implementation. WDYT?

Reply:
Indeed, good point. I pushed the changed in https://github.com/DataDog/dd-sdk-android/pull/1362/commits/8d45e9a30f51eeac882bae72c7fe619c553cc07a.